src/structures/interfaces/messageInteraction.ts

```diff
@@ -321,6 +321,44 @@ export class MessageInteractionContext {
     }
     return this.creator._componentCallbacks.delete(`${message_id}-${custom_id}`);
   }
+
+  /**
+   * Registers a wildcard component callback on a message.
+   * This unregisters automatically when the context expires.
+   * @param message_id The message ID of the component to register
+   * @param callback The callback to use on interaction
+   * @param expiration The expiration time of the callback in milliseconds. Use null for no expiration (Although, in this case, global components might be more consistent).
+   * @param onExpired A function to be called when the component expires.
+   */
+  registerWildcardComponent(
+    message_id: string,
+    callback: ComponentRegisterCallback,
+    expiration: number = 1000 * 60 * 15,
+    onExpired?: () => void
+  ) {
+    if (this.expired) throw new Error('This interaction has expired');
+    if (!this.initiallyResponded || this.deferred)
+      throw new Error('You must send a message before registering components');
```

Comment:
This isn't needed, since the message ID is given as an argument and callbacks aren't dependent on the parent context.

(and I should've changed this in other functions lol)

Reply:
https://github.com/Snazzah/slash-create/blob/9cdd970cc52c622c7d25da76dd5a6cb61915ece3/src/structures/interfaces/messageInteraction.ts#L302-L303

I would presume you refer to this as well (for `registerComponentFrom`, not `registerComponent`).